core/client/src/main/java/alluxio/client/block/stream/PacketOutStream.java

```diff
@@ -253,24 +253,26 @@ private void updateCurrentPacket(boolean lastPacket) throws IOException {
       return;
     }
 
-    if (mCurrentPacket.readableBytes() > 0) {
-      if (mCurrentPacket.writableBytes() == 0 || lastPacket) {
-        try {
+    if (mCurrentPacket.writableBytes() == 0 || lastPacket) {
```

Comment:
Since we already have an if clause for `lastPacket` afterward, would it make sense to move the `lastPacket` clean up there?